eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -8,9 +8,60 @@ param (
 )
 Set-StrictMode -Version 3
 
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.7 -Force -Scope CurrentUser
+
 . $PSScriptRoot/ChangedFiles-Functions.ps1
 . $PSScriptRoot/Logging-Functions.ps1
 
+function Find-Suppressions-Yaml {
+  param (
+    [string]$file
+  )
+
+  $currentDirectory = Get-Item (Split-Path -Path $file)
+
+  while ($currentDirectory) {
+    $suppressionsFile = Join-Path -Path $currentDirectory.FullName -ChildPath "suppressions.yml"
+    if (Test-Path $suppressionsFile) {
+      return $suppressionsFile
+    } else {
+      $currentDirectory = $currentDirectory.Parent
+    }
+  }
+
+  return $null
+}
+
+function Get-Suppression {
+  param (
+    [string]$file
+  )
+
+  $suppressionsFile = Find-Suppressions-Yaml $file
+  if ($suppressionsFile) {
+    $suppressions = Get-Content -Path $suppressionsFile -Raw | ConvertFrom-Yaml
+    foreach ($suppression in $suppressions) {
+      # Disable strict mode to handle tool or path not specified
+      Set-StrictMode -Off
+      $tool = $suppression.tool
+      $path = $suppression.path
+      Set-StrictMode -Version Latest
+
+      if ($tool -eq "TypeSpecRequirement") {
+        # Paths in suppressions.yml are relative to the file itself
+        $fullPath = Join-Path -Path (Split-Path -Path $suppressionsFile) -ChildPath $path
+
+        # If path is not specified, suppression applies to all files
+        if (!$path -or ($file -like $fullPath)) {
```

Comment:
Why use the like operator here? Is that to support basic wildcards?

Reply:
We want to support some form of wildcards.  Options:

1. globstar (`**`)
    * Used by `bash` and `git`
    * Node could use https://www.npmjs.com/package/glob
    * Couldn't find a PowerShell package implementing this
2. regex (`.*`)
    * Should be fairly consistent across languages
    * Relatively unusual to use for matching filenames
3. custom syntax (`*` means "zero or more chars")
    * PowerShell wildcards are pretty close but a superset of the needed feature (https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_wildcards?view=powershell-7.4)
    * Not sure how to implement in other languages

@weshaggard, @konrad-jamrozik: What do you think?  I'm leaning towards regex since I think it's the most standard across languages.

Reply:
I think I would lean towards regex as well. I don't want to have to implement a different globing/wildcard for each tool that needs to support these paths in a suppression file.

Reply:
Deferring this decision until we implement the shared component.  For now, I will use globstar `**` syntax in any `suppresions.yml` files I write.  A nice thing about `**` is PowerShell `-like` treats it exactly the same as `*`, so you can use `-like` to match input paths against a globstar pattern (may not be exactly equivalent, but close enough for our simple usage).